features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/internal/domain/event/BigIntegerUtilsTest.kt

```diff
@@ -0,0 +1,198 @@
+
+        // When
+        var mostSignificantAsHexString: String
+        val executionTime = measureNanoTime {
+            mostSignificantAsHexString = testedUtils.mostSignificant64BitsAsHex(traceId)
+        }
+
+        // Then
+        assertThat(mostSignificantAsHexString).isEqualTo(MIN_UNSIGNED_LONG_HEXA_STRING)
+        assertThat(executionTime).isLessThan(MAX_EXEC_TIME_IN_NANOS)
+    }
+
+    // endregion
+
+    // region leastSignificantUnsignedLongAsDecimal
+
+    @Test
+    fun `M use unsigned bits W extracting LSB from a max low trace id { decimal }`() {
+        // Given
+        val traceId = BigInteger(fakeHalfTraceIdHexaString + MAX_UNSIGNED_LONG_HEXA_STRING, 16)
+
+        // When
+        var leastSignificantAsDecimalString: String
+        val executionTime = measureNanoTime {
+            leastSignificantAsDecimalString = testedUtils.leastSignificant64BitsAsDecimal(traceId)
+        }
+
+        // Then
+        assertThat(
+            BigInteger(
+                leastSignificantAsDecimalString,
+                10
+            ).toString(16).padStart(16, '0')
+        ).isEqualTo(MAX_UNSIGNED_LONG_HEXA_STRING)
+        assertThat(executionTime).isLessThan(MAX_EXEC_TIME_IN_NANOS)
+    }
+
+    @Test
+    fun `M use unsigned bits W extracting LSB from a min low trace id { decimal }`() {
+        // Given
+        val traceId = BigInteger(fakeHalfTraceIdHexaString + MIN_UNSIGNED_LONG_HEXA_STRING, 16)
+
+        // When
+        var leastSignificantAsDecimalString: String
+        val executionTime = measureNanoTime {
+            leastSignificantAsDecimalString = testedUtils.leastSignificant64BitsAsDecimal(traceId)
+        }
+
+        // Then
+        assertThat(
+            BigInteger(
+                leastSignificantAsDecimalString,
+                10
+            ).toString(16).padStart(16, '0')
+        ).isEqualTo(MIN_UNSIGNED_LONG_HEXA_STRING)
+        assertThat(executionTime).isLessThan(MAX_EXEC_TIME_IN_NANOS)
+    }
+
+    @Test
+    fun `M extract unsigned less significant bits W extract { decimals }`() {
+        // Given
+        val traceId = BigInteger(fakeTraceIdAsHexaString, 16)
+
+        // When
+        var leastSignificantAsDecimalString: String
+        val executionTime = measureNanoTime {
+            leastSignificantAsDecimalString = testedUtils.leastSignificant64BitsAsDecimal(traceId)
+        }
+
+        // Then
+        assertThat(
+            BigInteger(
+                leastSignificantAsDecimalString,
+                10
+            ).toString(16).padStart(16, '0')
+        ).isEqualTo(fakeTraceIdAsHexaString.takeLast(16))
+        assertThat(executionTime).isLessThan(MAX_EXEC_TIME_IN_NANOS)
+    }
+
+    // endregion
+
+    companion object {
+        private const val MAX_UNSIGNED_LONG_HEXA_STRING = "ffffffffffffffff"
+        private const val MIN_UNSIGNED_LONG_HEXA_STRING = "0000000000000000"
+        private val MAX_EXEC_TIME_IN_NANOS = TimeUnit.MILLISECONDS.toNanos(8)
```

Comment:
why it is set to such value?

Reply:
well usually this is the max value you want to hit if you want to have 120 FPS. Maybe we can lower it even more ?

Reply:
it is not really relevant, because these are the unit tests, which are executed on JVM, on a server CPU, etc. The execution number you get here is be quite different from the one on the real device.